terraform/keycloak.tf

```diff
@@ -0,0 +1,157 @@
+locals {
+  user_list   = concat(var.reader_list, var.author_list, var.admin_list)
+}
+
+provider "keycloak" {
+  client_id     = var.kc_terraform_auth_client_id
+  client_secret = var.kc_terraform_auth_client_password
+  url           = var.kc_base_url
+  realm         = var.kc_realm
+}
+
+# Get Unique ID in keycloak
+data "keycloak_realm" "realm" {
+  provider = keycloak
+  realm    = var.kc_realm
+}
+
+data "keycloak_openid_client" "realm_management" {
+  provider = keycloak
+
+  realm_id  = var.kc_realm
+  client_id = var.kc_openid_client_id
+}
+
+data "keycloak_user" "this" {
+  provider = keycloak
+
+  for_each = toset(local.user_list)
+  realm_id = data.keycloak_realm.realm.id
+  username = each.key
+}
+
+
+## Creation of the roles
+resource "keycloak_role" "reader_role" {
+  provider = keycloak
+
+  realm_id  = data.keycloak_realm.realm.id
+  client_id = data.keycloak_openid_client.realm_management.id
+  name        = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:saml-provider/BCGovKeyCloak-tmhl5tvs,${aws_iam_role.quicksight_reader_role.arn}"
```

Comment:
`BCGovKeyCloak-tmhl5tvs` should be an input variable. This changes if keycloak is in a different realm.

Reply:
Yes I've changed it to var.kc_realm